demos/common/monitors/memory_monitor.h

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2019 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include<iostream> // TODO: remove
+
+#include <deque>
+
+class MemoryMonitor {
+public:
+    MemoryMonitor();
+    bool isEnabled() const;
+    void enable(std::size_t  historySize);
+    void disable();
+    void collectData();
+    std::deque<std::pair<double, double>> getLastHistory() const;
+    double getMeanMem() const;
+    double getMeanSwap() const;
+    double getMaxMem() const;
+    double getMaxSwap() const;
+
+    const double memTotal, swapTotal; // in GiB
```

Comment:
I don't think these can be treated as constants, as the former can change if the system has hotpluggable memory, and the latter can change if swap files/partitions are added/removed/resized.

Reply:
updated